components/brave_wallet/browser/internal/orchard_storage/orchard_storage.cc

```diff
@@ -502,12 +521,14 @@ ZCashOrchardStorage::GetSpendableNotes(const mojom::AccountIdPtr& account_id) {
   return base::ok(std::move(result));
 }
 
-std::optional<ZCashOrchardStorage::Error> ZCashOrchardStorage::UpdateNotes(
+std::optional<OrchardStorage::Error> OrchardStorage::UpdateNotes(
     const mojom::AccountIdPtr& account_id,
     const std::vector<OrchardNote>& found_notes,
     const std::vector<OrchardNoteSpend>& found_nullifiers,
```

Comment:
This may be something out of scope and to open an issue for, but these vectors should be just `span` of these types. We are having an extra unnecessary copy with:

```
  std::vector<OrchardNote> notes_to_add =
      block_scanner_results.discovered_notes;
  base::Extend(existing_notes.value(), notes_to_add);
```

But were this argument here a span, we could have done just:

```
base::Extend(existing_notes.value(), std::move(block_scanner_results.discovered_notes));
```

And then we could just slice the added arguments at the end of the vector, to create a span and call this function, avoiding one unnecessary copy of these vectors, which this code seems to do in a lot of cases that is not needed.

Reply:
Done